### PR TITLE
fix: Save and authorise flow in current editing environment

### LINKED ISSUE #26113: [Bug]: OAuth2.0 Staging configuration Save and Authorize is failing
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Configure Staging to connect to Dropbox with OAuth2.0, Save and Authorize - error comes up.

### Steps To Reproduce

1. Login as a user with access to Multiple Env. 
2. From New DS page, Click on Authenticated API
3. Now Go to Staging, and configure OAuth to connect to Dropbox. 
4. Fill all the details and click on Save and Authorize. 
User is taken to a screen displaying the following error.
{"responseMeta":{"status":404,"success":false,"error":{"code":"AE-APP-4027","title":"INTERNAL_ERROR","message":"Unable to find datasource 64d1d82e22c5f01dd568c747","errorType":"No resource found"}},"errorDisplay":"AE-APP-4027: Unable to find datasource 64d1d82e22c5f01dd568c747"}

https://github.com/appsmithorg/appsmith/assets/101863839/8993e459-9a34-4f53-b4ec-ef54f8a30372

Note: Same configuration works on Production

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ChandanBalajiBP**
#### PR fixes following issue(s)
Fixes #26113 

#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
>
- [x] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [x] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/sagas/DatasourcesSagas.ts [hunk 1]
```diff
@@ -2012,5 +2012,13 @@ export function* watchDatasourcesSagas() {
       ReduxActionTypes.SET_DATASOURCE_EDITOR_MODE,
       setDatasourceViewModeSaga,
     ),
+    takeEvery(
```
[the remark this answers is not in the document]
Comment (ChandanBalajiBP) [reply app/client/src/sagas/DatasourcesSagas.ts]: Surprisingly we have duplicate code in EE. This was a bad commit, have update it accordingly.

[the remark this answers is not in the document]
Comment (ChandanBalajiBP) [reply app/client/src/sagas/DatasourcesSagas.ts]: This duplication was introduced by code split and ME main PR.

**Review by ayushpahwa — APPROVED** [review]

**Comment by prapullc** [comment]
verified and LGTM we can move it to release

**Review by sneha122 — APPROVED** [review]

**Pull request merged**